src/Contract/Entity/TimestampableInterface.php

```diff
@@ -8,13 +8,27 @@
 
 interface TimestampableInterface
 {
+    /**
+     * Updates timestamps for createdAt and updatedAt properties.
+     */
     public function updateTimestamps(): void;
 
-    public function getCreatedAt(): DateTimeInterface;
-
-    public function setCreatedAt(DateTimeInterface $createdAt): void;
-
-    public function getUpdatedAt(): DateTimeInterface;
-
-    public function setUpdatedAt(DateTimeInterface $updatedAt): void;
+    /**
+     * Returns a DateTimeInterface object with the current timestamp.
+     */
+    public function getCurrentDateTime(): DateTimeInterface;
+
+    /**
+     * Returns an array of createdAt properties.
+     *
+     * @return string[]
+     */
+    public function getCreatedAtProperties(): array;
+
+    /**
+     * Returns an array of updatedAt properties.
+     *
+     * @return string[]
+     */
+    public function getUpdatedAtProperties(): array;
```

Comment:
In theory the changes to the traits could also break older installations, especially if one did not use the master trait but a sub-trait.

I could create a whole namespace for these files in `Timestampable` that are unrelated to the previous ones? Though that means duplicate implementations. I could mark the old one as deprecated,

Reply:
This is about interfaces, not traits.

If `TimestampableInterface` has a new method, all the classes that implement it will break.

Reply:
As long as those classes used the Trait it will not break.

If the classes used some of the traits it will still break, even if I create a separate Interface.

So just creating another Interface will not fully fix BC compatibility.

However, it will decrease the chance of it breaking, so I can do that.

Just wanted to make sure we are clear that it will not fully fix BC.

Reply:
> As long as those classes used the Trait it will not break.

That's an assumption. Only the interface is reliable source, everything else can be custom.
Any change of interface will break the contract. We can allow that in 3.0, but it's too soon. 2.0 was released just 6 months ago.

Reply:
Yes it is an assumption which is why I said just creating another interface will not be 100% backwards compatible either 
as one could have used one of the traits but modified one of the original methods which will not work, if I just create another interface.

I am quite sure it will break just as much as just coying the interface as traits are external usable files.

The only 100% solution would be to create a separate interface and separate traits. 

For isntance `TimestampableMethodsTrait` uses completely different methods than before.

Due to the different methods I will need to create separate traits anyway I think.

I have no issue with keeping the original as is and create subfolders in the timestampable parts like `Configurable` and the new traits / itnerfaces there.

That is the only way we can be sure regarding backwards compatibility and if you are fine with that I will change it accordingly. 

However, I do not want to just create a separate interface and still break backwards compatibility because people expect the traits to works as before. I just want to make sure you are fine with creating separate interfaces and traits.